crates/autopilot/src/run_loop.rs

```diff
@@ -429,6 +475,54 @@ impl RunLoop {
         Ok(())
     }
 
+    /// Tries to find a `settle` contract call with calldata ending in `tag`.
+    ///
+    /// Returns None if no transaction was found within the deadline or the task
+    /// is cancelled.
+    async fn wait_for_settlement_transaction(
```

Comment:
Is there a way to test this in e2e or unit test?

Reply:
Actually, I couldn't find a quick way to achieve that. That would require either an artificial delay in the driver's response or creating unit tests with mocks. Someone may have a better idea of how to test it.